packages/wrangler/src/api/startDevWorker/DevEnv.ts

```diff
@@ -8,18 +8,18 @@ import { ProxyController } from "./ProxyController";
 import { RemoteRuntimeController } from "./RemoteRuntimeController";
 import type { Controller, RuntimeController } from "./BaseController";
 import type { ErrorEvent } from "./events";
-import type { StartDevWorkerOptions, Worker } from "./types";
+import type { StartDevWorkerInput, Worker } from "./types";
 
 export class DevEnv extends EventEmitter {
 	config: ConfigController;
 	bundler: BundlerController;
 	runtimes: RuntimeController[];
 	proxy: ProxyController;
 
-	startWorker(options: StartDevWorkerOptions): Worker {
+	async startWorker(options: StartDevWorkerInput): Promise<Worker> {
```

Comment:
I didn't expect the async config updates to make startWorker async

We should expose a `async nextConfig()` and use the deferred pattern for resolvedConfig in ConfigController

I can add a commit to this PR to make this change for you if you want?